### PR TITLE
Support our new backups procedure

### PR DESCRIPTION
**Pull request by tomhukins**

### CODE REVIEW

File: lib/Brass/API.pm [hunk 1]
```diff
@@ -98,16 +99,12 @@ get 'api/pwd/' => sub {
...
 
     my $param = query_parameters->get('param');
-    $action eq 'sqldb' && !$param
-        and error __"Please specify required username for SQL password";
-    $action eq 'system' && !$param
-        and error __"Please specify required username for system password";
-    $action eq 'wazuh' && !$param
-        and error __"Please specify required username for wazuh password";
+    !$param
+        and error __"Please specify required username for $action password";
```
Comment (abeverley) [inline lib/Brass/API.pm]: As a general point, this is better written as ```__x"Please specify required username for {action} password", action => $action```. This abstracts potential future translations, not that that will even happen!

File: lib/Brass/Actions.pm [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        return @allowed_action;
+    }
+
+    my %is_allowed_action = map { $_ => undef } @allowed_action;
+    sub is_allowed_action {
+        my $action_name = $_[0];
+        return exists $is_allowed_action{$action_name};
+    }
+
+}
```
Comment (abeverley) [inline lib/Brass/Actions.pm:21]: Thanks Tom. Is there a reason for all this being within its own block?

File: lib/Brass/API.pm (hunk 1 above)
Comment (tomhukins) [reply lib/Brass/API.pm]: I'll change that to make it consistent with the rest of the code.

File: lib/Brass/Actions.pm (hunk 2 above)
Comment (tomhukins) [reply lib/Brass/Actions.pm:21]: Before I moved this to its own module, it was in `Brass::API` and I wanted to keep the variables private.  There's no good reason for it here, though, so I'll remove the block.

**Pull request merged**